spinnaker/spinnaker_system/google_http_lb_upsert_server_test.py

```diff
@@ -0,0 +1,195 @@
+  '''Test fixture for Http LB test.
+  '''
+
+  FIRST_CERT = ''
+  SECOND_CERT = ''
+
+
+  @staticmethod
+  def make_ssl_cert(name, bindings):
+    '''Create and return an SslCertificate in dictionary format.
+    '''
+    key = crypto.PKey()
+    key.generate_key(crypto.TYPE_RSA, 2048)
+    cert = crypto.X509()
+    cert.get_subject().C = "US"
+    cert.get_subject().ST = "New York"
+    cert.get_subject().L = "New York City"
+    cert.get_subject().O = "Google"
+    cert.get_subject().OU = "Spinnaker"
+    cert.get_subject().CN = "localhost"
+    cert.set_serial_number(4096)
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(24 * 60 * 60)
+    cert.set_issuer(cert.get_subject())
+    cert.set_pubkey(key)
+    cert.sign(key, 'sha1')
+
+    return {
+      'name': name,
+      'privateKey': crypto.dump_privatekey(crypto.FILETYPE_PEM, key),
+      'certificate': crypto.dump_certificate(crypto.FILETYPE_PEM, cert)
+    }
+
+
+  @classmethod
+  def setUpClass(cls):
+    runner = citest.base.TestRunner.global_runner()
+    bindings = runner.bindings
+    cls.FIRST_CERT = 'first-cert-%s' % (bindings['TEST_APP'])
+    cls.SECOND_CERT = 'second-cert-%s'  % (bindings['TEST_APP'])
+    runner = citest.base.TestRunner.global_runner()
+    bindings = runner.bindings
+    scenario = runner.get_shared_data(GoogleHttpLoadBalancerTestScenario)
+    managed_region = runner.bindings['TEST_GCE_REGION']
+    title = 'Check Quota for {0}'.format(scenario.__class__.__name__)
+
+    verify_results = gcp.verify_quota(
+      title,
+      scenario.gcp_observer,
+      project_quota=GoogleHttpLoadBalancerTestScenario.MINIMUM_PROJECT_QUOTA,
+      regions=[(managed_region,
+                GoogleHttpLoadBalancerTestScenario.MINIMUM_REGION_QUOTA)]
+    )
+    if not verify_results:
+      raise RuntimeError('Insufficient Quota: {0}'.format(verify_results))
+
+    context = ExecutionContext()
+    compute_agent = gcp.GcpComputeAgent.make_agent(scopes=SCOPES,
+                                                   credentials_path=bindings['GCE_CREDENTIALS_PATH'])
+    compute_agent.invoke_resource(context, # No predicates against this agent, context is empty.
+                                  'insert',
+                                  resource_type='sslCertificates',
+                                  project=bindings['GCE_PROJECT'],
+                                  body=cls.make_ssl_cert(cls.FIRST_CERT, bindings))
+    compute_agent.invoke_resource(context,
+                                  'insert',
+                                  resource_type='sslCertificates',
+                                  project=bindings['GCE_PROJECT'],
+                                  body=cls.make_ssl_cert(cls.SECOND_CERT, bindings))
+
+
+  @classmethod
+  def tearDownClass(cls):
+    runner = citest.base.TestRunner.global_runner()
+    bindings = runner.bindings
+    context = ExecutionContext()
+    compute_agent = gcp.GcpComputeAgent.make_agent(scopes=SCOPES,
```

Comment:
I'm not sure I can do that from the cleanup/teardown methods since they are class methods. There's no `self` to access the scenario. I couldn't find any workarounds, do you have any ideas?